source/features/show-open-prs-of-forks.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {isRepoWithAccess} from '../libs/page-detect';
+import {getUsername} from '../libs/utils';
+
+function findForkedRepo(): string | undefined {
+	const forkSourceElement = select<HTMLAnchorElement>('.fork-flag a');
+	if (forkSourceElement) {
+		return forkSourceElement.pathname.slice(1);
+	}
+
+	return undefined;
+}
+
+interface SearchResponse {
+	issueCount: number;
+	edges: Array<{
+		node: {
+			url: string;
+		};
+	}>;
+}
+
+function getUserPullRequestsURL(forkedRepo: string, user: string): string {
+	return `https://github.com/${forkedRepo}/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc+author%3A${user}`;
+}
+
+async function getOpenPullRequestsData(): Promise<void | SearchResponse> {
+	if (!isRepoWithAccess()) {
+		return;
+	}
+
+	const forkedRepo = findForkedRepo();
+
+	if (!forkedRepo) {
+		return;
+	}
```

Comment:
Agree on the need of caching. What would be the benefit of using `forkedRepo` as a parameter instead of just using memoization on `findForkedRepo`?

Reply:
I think it makes sense to have functions as “pure” as possible, i.e. not read from global state. `cache.function` itself is a memoization function, so its first parameter will be used as key of the cache.

Example here: https://github.com/sindresorhus/refined-github/pull/2868/files#diff-5664f505bec7f184c9588cc203936104R90

A future update to my cache module would simplify this code even more and it would make more sense to have parameters

Reply:
I generally agree about the purity aspect of functions, but I don't think it's too realistic in this extension as basically all information comes from the environment. I could pass the `forkedRepo` as a parameter, but I'd still rely on `isRepoWithAccess()` and `getUsername()`.

But the example you linked has the same problem. What was the reasoning there for using `latestTag` as a parameter, but not the other data?

Reply:
> but I'd still rely on isRepoWithAccess() and getUsername().

Not isRepoWithAccess, that check should be outside. 

getUsername is basically a constant for the whole session, the cache doesn’t depend on it. 

> But the example you linked has the same problem. What was the reasoning there for using latestTag as a parameter, but not the other data?

latestTag is the cache key, the other data isn’t (or is already part of it). Improvements may always be possible, but we should look at the lowest common denominator.

Reply:
Sure, was just trying to dig deeper to gain some insight into prior art decision making. 👍